tests/trainer/test_scale_schedule.py

```diff
@@ -91,21 +99,30 @@ def test_epochs_scaled(
             composer_trainer_hparams.algorithms = [ScaleScheduleHparams(ratio=ssr)]
         else:
             composer_trainer_hparams.scale_schedule_ratio = ssr
-        trainer = composer_trainer_hparams.initialize_object()
 
-        assert trainer.state.max_duration.unit == TimeUnit.EPOCH
-        assert trainer.state.max_duration.value == int(10 * ssr)
-        scheduler = trainer.state.schedulers[0]
+        class CheckScaleSchedule(Callback):
```

Comment:
Should prob pull this out and make it a private class in the file rather than an inner class for this one test

In general best to avoid nested classes